lib/project_types/extension/cli.rb

```diff
@@ -62,17 +63,28 @@ module Argo
   end
 
   module Models
+    module SpecificationHandlers
+      autoload :Default, Project.project_filepath('models/specification_handlers/default')
+    end
+
     autoload :App, Project.project_filepath('models/app')
     autoload :Registration, Project.project_filepath('models/registration')
     autoload :Version, Project.project_filepath('models/version')
-    autoload :Type, Project.project_filepath('models/type')
     autoload :ValidationError, Project.project_filepath('models/validation_error')
-
-    class << self
-      Models::Type.load_all
-    end
+    autoload :Specification, Project.project_filepath('models/specification')
+    autoload :Specifications, Project.project_filepath('models/specifications')
   end
 
   autoload :ExtensionProjectKeys, Project.project_filepath('extension_project_keys')
   autoload :ExtensionProject, Project.project_filepath('extension_project')
+
+  def self.specifications
```

Comment:
Having this up on the autoload level does feel a bit odd. Was there something that required this to be here?

Reply:
I put it there because we don't have a dedicated file that defines methods on `Extension`. I can extract this code into a separate file as long as I won't run into any auto loading related issues.

Reply:
Given that the main entry point for a project type is `<project-type>/cli.rb`, which in our case defines `Extension`, I'll have to leave the code here as autoloading won't trigger when the module is already defined.